### ISSUE TITLE
Possible feature update: Only spawn from actual trees

### ISSUE THREAD

**Issue by BrekiTomasson** [RandomMcSomethin/fallingleaves#10]
This should be configurable with an on/off toggle, but somehow it feels wrong when I place a leaf block manually, that leaves begin falling from it. There should be some kind of check (similar to the leaf-decay check) which checks if the leaf block is in concact with a log (or within touching distance of other leaf blocks that touch a log). I'm sure there are already samples of code available that do this very check that we can slot in.

**Comment by Fourmisain** [comment]
I just found `state.get(LeavesBlock.PERSISTENT)` which is true exactly if it was placed manually.
There's also `state.get(LeavesBlock.DISTANCE)`, an integer between 1 and 7 telling the distance to an appropriate log.

Leaves despawn when `state.get(DISTANCE) == 7 && !state.get(PERSISTENT)`.

So we could either use the distance or the persistence flag.
I guess it makes more sense to use the persistence flag, since most of the times, leaves are placed nearby logs anyways, so there wouldn't be an actual change.

How should we name the config option?
"Drop from manually placed leaves", defaulting to true?
It's the shortest description I can come up with and still kinda a mouthfull

**Comment by BrekiTomasson** [comment]
I think `LeavesBlock.DISTANCE` returns 0 if you've got a leaf block touching a stone block touching a log. It should only return the distance as it travels along other leaf blocks, making it a good way to check if a leaf block is part of an actual tree or part of something else. I haven't checked if it only works in straight lines and 90-degree angles or if it supports diagonal distances, as that would be required for some modded tree types.

As for the config option, I'd probably prefer something like "Only drop from actual trees" or similar, defaulting to false. Yours is good too, although I agree that "manually" might make it a bit wordy. Maybe "Drop from player-placed leaves"?

**Comment by BrekiTomasson** [comment]
I believe this issue can be closed?

**Comment by Fourmisain** [comment]
For me, the "player placed" option is enough, though it is different from what was originally suggested.
Implementing an actual tree check would be too expensive and basically impossible to get 100% right anyway.
There might be some merit of looking at `LeavesBlock.DISTANCE` but I'm not sure if it's worth getting into.

So I'd vote close until someone has a better suggestion.

**Comment by BrekiTomasson** [comment]
Agreed, the "Player Placed" option should be enough. We can revisit this at a later date to see how leaf blocks placed as part of structures in Worldgen are handled.
